merge.go

```diff
@@ -515,3 +516,50 @@ func normalizeString(ctx context, opts *options, str string) (value, Error) {
 
 	return newSplice(ctx, opts.meta, varexp), nil
 }
+
+func fieldOptsOverride(opts *options, fieldName string) *options {
+	if opts.fieldValueHandling == nil {
+		return opts
+	}
+	cfgHandling, ok := opts.fieldValueHandling[fieldName]
+	if !ok {
+		// possible unknown depth defined
+		cfgHandling, ok = opts.fieldValueHandling[fmt.Sprintf("**.%s", fieldName)]
+	}
+	if !ok {
+		if len(opts.fieldValueHandling) != 0 {
+			// need to remove a level of fields as a nested field could have a
+			// config handling modification
+			newOpts := new(options)
+			*newOpts = *opts
+			newOpts.fieldValueHandling = trimFieldConfigHandlingLevel(fieldName, newOpts.pathSep, newOpts.fieldValueHandling)
+			opts = newOpts
```

Comment:
nit; one only needs `new` very rarely. Common shallow-copy pattern is easily expressed like this:

```
  newOpts := *opts
  newOpts.<settings> = ...
  return &newOpts
```